File: problems/0121-best-time-to-buy-and-sell-stock/solution.c
int maxProfit(int* prices, int pricesSize) {
    if (pricesSize == 0) return 0;  // No prices available

    int min_price = prices[0];      // Start with the first price
    int max_profit = 0;             // Initial maximum profit is 0

    for (int i = 1; i < pricesSize; i++) {
        if (prices[i] < min_price) {
            min_price = prices[i];  // Update the minimum price
        }

        int profit = prices[i] - min_price;  // Calculate the profit from the current price
        if (profit > max_profit) {
            max_profit = profit;  // Update the maximum profit if we found a better one
        }
    }

    return max_profit;
}

File: problems/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution(object):
    def maxProfit(self, prices):
        n=2**31-1
        max_profit=0
        for i in range (len(prices)):
            if prices[i]<n:
                n=prices[i]

            diff=prices[i]-n
            max_profit=max(diff,max_profit)
        return max_profit        
        
